src/cert/ssl_cert_provider.cpp

```diff
@@ -109,7 +109,14 @@ long random_long()
 {
     long out{0};
     std::array<uint8_t, 4> bytes;
+
+    // The array must be used uninitialized. Otherwise, there will be no randomness.
+#ifndef __clang__
```

Comment:
This needs to account for MSVC and AppleClang, right? I believe we can do that by checking `__GNUC__` too. 

It seems that [other compilers define that too](https://stackoverflow.com/questions/28166565/detect-gcc-as-opposed-to-msvc-clang-with-macro#comment96299743_43205345). So ideally we'd set `-DMULTIPASS_COMPILER_GCC` in `CMakeLists.txt` and then check that here (similar to what we do for `MULTIPASS_PLATFORM_LINUX/WINDOWS/APPLE`). But if you want to check just `(defined(__GNUC__) && !defined(__clang__))`, I guess that'd be OK too.

Reply:
Hey @ricab, thanks for reviewing this!

Yes, the issue here is that it must _only_ work with gcc and, as you mention, other compilers do emulate the `#pragma GCC`. The point is that the block `#ifndef __clang__`...`#endif` also contains code :man_facepalming:. I'd fix that first. Let me give it a try.

Reply:
Done, let's see now...

Reply:
Oh indeed, I missed that entirely.